### DISCUSSION TITLE
Do we have NextJs @12.X.X version support for this Module Federation Plugin?

### DISCUSSION THREAD

**Discussion by 12dkaskar** [webpack/webpack#16026] [General]
# Bug report

**What is the current behavior?**

**Plugin:** Webpack Module Federation

**Next JS Version:** 12.1.6

**Host App :** Build and stand alone App is working as expected

**Micro App:** Here's the issue, Remote entry file that is generated during the build process does not contain the exposed modules. Below is the sample of Remote Entry File.

`"use strict";
var moduleMap = {
	"./blogPage": function() {
		var e = new Error("Cannot find module './templates/BlogPage'"); e.code = 'MODULE_NOT_FOUND'; throw e;
	},
	"./articlePage": function() {
		var e = new Error("Cannot find module './templates/ArticlePage/ArticlePage'"); e.code = 'MODULE_NOT_FOUND'; throw e;
	},
	"./tagPage": function() {
		var e = new Error("Cannot find module './templates/TagPage'"); e.code = 'MODULE_NOT_FOUND'; throw e;
	},
}`

**Error Observed:** Cannot Find Module
Where is with NextJS 11.X.X Version the remote entry file is loading all the components correctly which are exposed by the microapp.

**What is the expected behavior?**

Remote Entry File generated after the build should contain successfully loading of the modules which are been exposed by the micro app in Microfrontend Architecture.

**Other relevant information:**
webpack version:  5.66.0
Node.js version: 16.13.1
Operating System:  Windows 10
Additional tools: NA

**Comment by vankop** [comment]
`MODULE_NOT_FOUND` means there is no such module. you probably get more informative error in stdout..
________

Maybe you have change directory structure and now there is no such modules?

**Comment by ScriptedAlchemy** [comment]
Next.js does not work with webpack federation out the box. It requires third party implementations. 

Nextjs-MF offers total support for next and federation.
